### PR TITLE
First draft of view life cycle lab

### PR DESCRIPTION
**Pull request by mdippery**
Looking forward to some feedback on this one. I'm a bit afraid I didn't provide quite enough direction, so I'd like to hear your thoughts on whether this is the right direction/format for the lab or not.

I didn't write the UI tests yet, either. Have to look into more details on how specifically to do that, and I wanted some feedback on the README before I dive deeply into the UI tests. (I'm a bit behind on the deadline for this unit and I want to keep rolling for the time being.)

cc: @AnnJohn @JimCampagno

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,32 +22,78 @@
...
+2. Open up `Main.storyboard` to start designing your interface.
+3. Open up `ViewController.swift` in an Assistant Editor pane.
+4. You should first add the six views representing the dice to the top half of the interface. You can drag these views from the object library onto the Interface Builder canvas, and arrange them as shown in the screenshots above.
+5. You should then add the six colored score labels along the middle of the screen. Remember: You can set their background color using the Attributes Inspector.
+6. Since you will be interacting with all twelve of these views, you should configure IB outlets in `ViewController` so that the view controller can interact with them. <kbd>Control</kbd>-drag a connection from each of these twelve views to `ViewController.swift` to create them.
+7. These views should initially be hidden from the user, so you should do some setup in `ViewController`'s `viewDidLoad()` method to hide them initially.
+8. The main view already has a button configured at the bottom, but it is not hooked up to anything yet. <kbd>Control</kbd>-drag from the button to `ViewController.swift` to create a new IB outlet for the button.
```
Comment (JimCampagno) [inline README.md]: Need to change IB outlet to IB action. I can do this as well, making note.

File: README.md [hunk 2]
```diff
@@ -23,32 +22,78 @@
...
+9. The die button's IB outlet needs to do several things:
```
Comment (JimCampagno) [inline README.md]: Same here, needs to be IB action instead.

**Comment by JimCampagno** [comment]
Overall, a really great lab. 

The screenshots you provided were taken on an iPhone 6s plus (which is OK for now - NO Action required on your end), this is the same problem I outlined in that prior lesson. When the student clones down the project, it will default to iPhone 6s plus. I will spend time to see if we can have it default to iPhone 5S when they clone the project down, or we could just tell them to change it themselves and be done with it.
iPhone 5S Simulator

---

This is the **perfect** direction/format to take the student in. As this will be one of the first real tough lessons, taking them through what in the end it should look like and how it should function is key (and you did that). If they can see the path at some level, they might be able to hack something together (albeit it might be very ugly).

In later lessons, you probably could be even shorter in your explanation (similar to the lessons I provided below). Just providing screenshots to show them the path could suffice in future lessons, but still providing instructions. This all depends of course on the problem at hand... a good sense of how lessons in the past were written (used in the immersive course) can be seen here. I actually think the approach I want to take (and what you did here) is better!... but that being said you might not have all the time in the world and I want you to find a good rhythm.

To be honest, if every lesson was like this.. as far as your outline/instructions that would be AWESOME but I also don't want you to kill yourself with the time you do have.. where if you want, you can do what I said in the prior paragraph (or ignore it if you have the time to be this detailed).

[Swift Boat](https://github.com/learn-co-curriculum/swift-boat)
[Array Practice-OBJ-C](https://github.com/learn-co-curriculum/objc-array-practice)

You can forget about writing the UI tests for this unit. I'm less a fan of testing UI.. and I don't want to force that, especially in a lesson like this which might have students creating some insane code to get the job done.

@mdippery

**Comment by JimCampagno** [comment]
Last thing to note. My outlines for the final three labs in this unit were vague. We could make those much easier and re-arrange the labs here... having this one be the final one. Feel free to be creative as well on a lab you want to make(where it doesn't have to be as involved as this one). I welcome all feedback!

@mdippery

**Comment by mdippery** [comment]
Thanks for the feedback! I made the IB outlet -> IB action corrections.

**Pull request merged**